test/1_stdlib/Tuple.swift

```diff
@@ -0,0 +1,110 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var TupleTestSuite = TestSuite("Tuple")
+
+// Test tuple comparison operators
+// all the tuple types use the same basic implementation for the operators
+// so testing any arity tests them all
+
+func testEquality<A: Equatable, B: Equatable, C: Equatable>(
+  lhs: (A,B,C), equal: Bool, to rhs: (A,B,C),
+  file: String = __FILE__, line: UInt = __LINE__
+) {
+  if equal {
+    expectTrue(lhs == rhs, file: file, line: line)
+    expectFalse(lhs != rhs, file: file, line: line)
+  } else {
+    expectFalse(lhs == rhs, file: file, line: line)
+    expectTrue(lhs != rhs, file: file, line: line)
+  }
+}
+
+TupleTestSuite.test("Tuple/equality") {
```

Comment:
Yeah, that would be nice. Though FWIW `checkComparable` doesn't handle the case of NaN (which is reasonable in that `Comparable` is strict total order which NaN violates, but still good to check).